pkg/handlers/v1/subnet.go

```diff
@@ -0,0 +1,77 @@
+package v1
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+type subnetConfiguration struct {
+	CIDRBlock string `json:"cidrBlock"`
+}
+
+type subnetConfigurationDiff struct {
+	PreviousValue *subnetConfiguration `json:"previousValue"`
+	UpdatedValue  *subnetConfiguration `json:"updatedValue"`
+	ChangeType    string               `json:"changeType"`
+}
+
+type subnetTransformer struct{}
+
+func (t subnetTransformer) Create(event awsConfigEvent) (Output, bool, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, false, err
+	}
+
+	var config subnetConfiguration
+	if err := json.Unmarshal(event.ConfigurationItem.Configuration, &config); err != nil {
+		return Output{}, false, err
+	}
+
+	change := extractSubnetInfo(&config)
+	change.ChangeType = added
+	output.Changes = append(output.Changes, change)
+	return output, false, nil
+}
+
+func (t subnetTransformer) Update(event awsConfigEvent) (Output, bool, error) {
```

Comment:
I believe so. I have not been able to find an example event where the CIDR block changes. The only AWS doc I've been able to find speaking to it is this one: https://aws.amazon.com/premiumsupport/knowledge-center/vpc-ip-address-range/

> It's not possible to change or modify the IP address range of an existing virtual private cloud (VPC) or subnet.

I _think_ this article confirms that the blocks can't change, but it may be worth some additional investigation.

Reply:
I'm convinced by that :)

Reply:
Yeah, I can confirm these CIDR blocks can't change once you have created a VPC. During creation there is an option to configure your own CIDR range, we cant change it post creation. I ran into hard blockers when I was working on configuring VPC peering in my previous team.